### PR TITLE
Mark Ipv4Addr is_unspecified as stable and provide reference.

### PR DESCRIPTION
**Pull request by therealbstern**
Per [#27709 (comment)](https://github.com/rust-lang/rust/issues/27709#issuecomment-231280999), no RFC is needed here.

IPv4 "unspecified" has been defined in [Stevens], and has been part of the IPv4 stack for quite some time.  This property should become stable, since this use of 0.0.0.0 is not going anywhere.

[Stevens][_UNIX Network Programming Volume 1, Second Edition_.  Stevens, W. Richard.  Prentice-Hall, 1998.  p. 891]

Please let me know if I got the rustdoc wrong or something.  I tried to be as terse as possible while still conveying the appropriate information.

This also has a slight impact on PR #34694, but that one came first, so this shouldn't block it, IMO.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR @therealbstern! Currently though the way the libs team handles stabilizations is to do so with a "final comment period" where we have an entire cycle to discuss whether an API should be stable or not. We'll actually nominate this coming Monday the set of APIs we'd like to see stabilized in 1.12.

Along those lines I've made sure to nominate the relevant issue (so we'll see it), and could you back out the changes to the stability tag here? Other than that the doc change looks good to me to merge!

**Comment by therealbstern** [comment]
Done and done.  Thank you!

**Comment by alexcrichton** [comment]
Could you also squash the commits together?

**Comment by alexcrichton** [comment]
Also, out of curiosity, is there any online sources of this information we could link to as well?

**Comment by therealbstern** [comment]
I put in a link to man7.  I didn't squash because I've already pushed to github, so it's rejecting the squashing.  Should I close this PR and open a new one, or leave it as is?

BTW, the documentation (and the welcome message from rust-highfive) implies not to squash.  Is this incorrect?

**Comment by brson** [comment]
@therealbstern You should be able to push the squashed branch to your repo with `-f`. Generally, we don't like squashing *during* review, because it makes it hard to follow review progress, but often reviewers ask for a squash before the final commit.

**Comment by therealbstern** [comment]
Thanks for the hint about -f -- squashed.
